packages/server/test/unit/modes/run_spec.js

```diff
@@ -522,6 +495,59 @@ describe('lib/modes/run', () => {
       })
     })
 
+    it('logs warning and resolves on failed video end', async function () {
+      process.nextTick(() => {
+        return this.projectInstance.emit('end', {
+          stats: {
+            failures: 0,
+          },
+        })
+      })
+
+      sinon.spy(videoCapture, 'process')
+      const endVideoCapture = sinon.stub().rejects()
+
+      await runMode.waitForTestsToFinishRunning({
+        project: this.projectInstance,
+        videoName: 'foo.mp4',
+        compressedVideoName: 'foo-compressed.mp4',
+        videoCompression: 32,
+        videoUploadOnPasses: true,
+        gui: false,
+        endVideoCapture,
+      })
+
+      expect(errors.warning).to.be.calledWith('VIDEO_POST_PROCESSING_FAILED')
+
+      expect(videoCapture.process).not.to.be.called
```

Comment:
to check my understanding: previously, this error would crash the process, now it's dressed up in a warning?

so e2e tests *can* still fail, but only in these 2 (more rare than now) conditions:

1. `WAIT_FOR_MORE_FRAMES_TIMEOUT` is exceeded and <2 frames are received; or
2. `ffmpeg` gives an error *not* caused by receiving 0 frames

is that all correct?

Reply:
the error was always a warning, but it would cause our tests to fail because the stdout didnt match when we printed the warning message.

yes those scenarios would cause stdout to show the warning and e2e tests to fail